sql/lease.go

```diff
@@ -151,11 +152,13 @@ func (s LeaseStore) Acquire(txn *client.Txn, tableID ID, minVersion DescriptorVe
 		}
 		return nil
 	})
```

Comment:
I see. I think a `roachpb.StripErrorTransaction()` function would be clearer than going through GoError/NewError (and I think doing this in db.Txn makes sense). Or maybe in the longer term we should move away from type-switching for error handling, and instead of asking "is this a transaction restart error", ask "is this a transaction restart error *for this transaction*". That way if an error does leak across transaction boundaries it would be treated as any opaque error.